tests/components/nordpool/conftest.py

```diff
@@ -44,40 +38,78 @@ async def load_int(
 
     config_entry.add_to_hass(hass)
 
-    with (
-        patch(
-            "homeassistant.components.nordpool.coordinator.NordPoolClient.async_get_delivery_period",
-            return_value=get_data,
-        ),
-    ):
-        await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.config_entries.async_setup(config_entry.entry_id)
     await hass.async_block_till_done()
 
     return config_entry
 
 
-@pytest.fixture(name="get_data")
+@pytest.fixture(name="get_client")
 async def get_data_from_library(
-    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker, load_json: dict[str, Any]
-) -> DeliveryPeriodData:
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
+    load_json: list[dict[str, Any]],
+) -> AsyncGenerator[NordPoolClient]:
     """Retrieve data from Nord Pool library."""
-
+    aioclient_mock.request(
+        "GET",
+        url=API + "/DayAheadPrices",
+        params={
+            "date": "2024-11-05",
+            "market": "DayAhead",
+            "deliveryArea": "SE3,SE4",
+            "currency": "SEK",
+        },
+        json=load_json[0],
+    )
+    aioclient_mock.request(
+        "GET",
+        url=API + "/DayAheadPrices",
+        params={
+            "date": "2024-11-05",
+            "market": "DayAhead",
+            "deliveryArea": "SE3",
+            "currency": "EUR",
+        },
+        json=load_json[0],
+    )
+    aioclient_mock.request(
+        "GET",
+        url=API + "/DayAheadPrices",
+        params={
+            "date": "2024-11-04",
+            "market": "DayAhead",
+            "deliveryArea": "SE3,SE4",
+            "currency": "SEK",
+        },
+        json=load_json[1],
+    )
+    aioclient_mock.request(
```

Comment:
Is there a reason we can't mock away the entire client to avoid having to cnonfigure `aioclient` requests here? Imo it's cleaner if we just mock the methods of the client and leave the request mocking to the lib tests

Reply:
We can. It's just much easier and cleaner to mock the responses instead of constructing the dataclasses manually for all the requests.
I see what you mean, but would prefer to leave it like this.